mytimer/functions.py

```diff
@@ -459,16 +459,28 @@ def run_timer(args):
         show_faces_list()
     elif args.programs_list:
         show_programs_list()
-    elif args.program == "pomodoro":
-        short_break_params = load_params(args, program="pomodoro-short-break", is_break=True)
-        long_break_params = load_params(args, program="pomodoro-long-break", is_break=True)
-        pomodoro_timer(
-            timer_func,
-            params=params,
-            long_break_params=long_break_params,
-            short_break_params=short_break_params)
-    elif args.program in ["52-17", "112-26", "animedoro"]:
-        break_params = load_params(args, is_break=True)
-        two_step_timer(timer_func, params1=params, params2=break_params)
     else:
-        timer_func(**params)
+        timer_round = 0
+        while timer_round < args.repeat or args.repeat == -1:
+            if args.program == "pomodoro":
+                short_break_params = load_params(args, program="pomodoro-short-break", is_break=True)
+                long_break_params = load_params(args, program="pomodoro-long-break", is_break=True)
+                pomodoro_timer(
+                    timer_func,
+                    params=params,
+                    long_break_params=long_break_params,
+                    short_break_params=short_break_params)
+            elif args.program in ["52-17", "112-26", "animedoro"]:
+                break_params = load_params(args, is_break=True)
+                two_step_timer(timer_func, params1=params, params2=break_params)
+            else:
+                timer_func(**params)
+
+            end_round_message = END_ROUND_MESSAGE.format(f'{timer_round + 1}/{args.repeat}')
+            if args.repeat == -1:
+                end_round_message = END_ROUND_MESSAGE.format(f'{timer_round + 1}')
+            print_message(
```

Comment:
I suggest we skip this line for the final round of the loop. Do you agree?

Reply:
Fixed in [fa7a0c4](https://github.com/sepandhaghighi/mytimer/pull/95/commits/fa7a0c4deff22df0b6ec8b6088373a3a54d340d6).